dpath/__init__.py

```diff
@@ -45,7 +49,22 @@ def _split_path(path: Path, separator: Optional[str] = "/") -> Union[List[PathSe
     else:
         split_segments = path.lstrip(separator).split(separator)
 
-    return split_segments
+    final = []
+    for segment in split_segments:
+        if (options.DPATH_ACCEPT_RE_REGEXP and isinstance(segment, str)
+                and segment[0] == '{' and segment[-1] == '}'):
+            try:
+                rs = segment[1:-1]
+                rex = re.compile(rs)
+            except Exception as reErr:
```

Comment:
`re.compile` throws a few exceptions but most of them are for improper usage and they're not relevant here.

The only case we would be interested in exceptions here is if the regex passed from the user is malformed. In this case `re.compile` will throw a `re.error`. I think it would be preferable if you caught that exception specifically for clarity.

Reply:
OK, added InvalidRegex exception